src/pip/_internal/configuration.py

```diff
@@ -420,3 +420,7 @@ def _mark_as_modified(self, fname, parser):
         file_parser_tuple = (fname, parser)
         if file_parser_tuple not in self._modified_parsers:
             self._modified_parsers.append(file_parser_tuple)
+
+    def __repr__(self):
+        # type: () -> str
+        return "{}({!r})".format(self.__class__.__name__, self._dictionary)
```

Comment:
There's more things that represent a Configuration object (eg. parsers, load_only etc) which we should probably include in the `repr`.

Reply:
I assumed this PR was what was helpful for debugging at some point. I would rather not preemptively populate any `__repr__`s.

Reply:
🤷‍♂ YAGNI applies and we can revisit this if we need to later.